src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,105 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int DEFAULT_SIZE = 0;
+    private int quantity = DEFAULT_SIZE;
+    private Object[] elements = new Object[DEFAULT_CAPACITY];
+
     @Override
     public void add(T value) {
-
+        growIfArrayFull();
+        elements[quantity] = value;
+        quantity += 1;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkInRangeForAddition(index);
+        growIfArrayFull();
+        System.arraycopy(elements, index, elements, index + 1, quantity - index);
+        elements[index] = value;
+        quantity += 1;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkInRangeForGet(index);
+        return (T) (elements[index]);
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkInRangeForGet(index);
+        elements[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkInRangeForGet(index);
+        Object temp = elements[index];
+
+        if (index != quantity - 1) {
+            System.arraycopy(elements, index + 1, elements, index, quantity - index + 1);
+        }
+        quantity -= 1;
+        return (T) (temp);
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < quantity; i++) {
+            if (elements[i] == null && element == null) {
+                return remove(i);
+            }
+            if (element != null && elements[i] == null) {
+                continue;
+            }
+            if (elements[i].equals(element)) {
+                return remove(i);
+            }
+        }
+        throw new NoSuchElementException("No such element present");
     }
 
     @Override
     public int size() {
-        return 0;
+        return quantity;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size() == 0;
+    }
+
+    private void growIfArrayFull() {
+        if (size() == elements.length) {
+            Object[] temp = new Object[elements.length + elements.length / 2];
```

Comment:
The size of the new array is 'elements.length + elements.length / 2'. This can lead to a problem if the length of 'elements' is 1 or 2 because the size of the new array will be the same as the old one. It's better to add 1 to the result of the division to ensure that the size of the new array is always greater than the old one.